week1_day2/exercise/3.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Transaction struct {
+	id  int32
+	t   int32 // 0: credit, 1: debit
+	amt int32
+}
+
+type Account struct {
+	mu      sync.Mutex
+	balance int32
+}
+
+func (a *Account) withdraw(amt int32) {
```

Comment:
this function should return golang error. Print the error at call site.

Reply:
Done. PTAL.